src/foam/dao/AbstractDAO.java

```diff
@@ -61,6 +61,13 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
     return sink;
   }
 
+  protected Sink decorateSink_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
```

Comment:
Just move the body of the other decorateSink_ (the one without the X) here and remove the other.